[FEATURE] - run a job on the last weekday of the month

### Describe the bug

Gocron cronjob cant parse "LW", errors with invalid syntax

### To Reproduce


`
package main
import (
	"log"
	"github.com/go-co-op/gocron/v2"
)
func main() {
	s, err := gocron.NewScheduler()
	if err != nil {
		log.Fatal(err)
	}

	_, err = s.NewJob(
		gocron.CronJob("0 0 5 LW * ?", true),
		gocron.NewTask(func() {
			log.Println("Test job 1")
		}),
	)
	if err != nil {
		log.Printf("Test job 1 err: %v", err)
	}

	_, err = s.NewJob(
		gocron.CronJob("0 5 LW * ?", false),
		gocron.NewTask(func() {
			log.Println("Test job 2")
		}),
	)
	if err != nil {
		log.Printf("Test job 2 err: %v", err)
	}

	s.Start()
	for {
	}
}
`

### Version

v2.13.0

### Expected behavior

The expression should be parsed since its a valid cron expression

I would love to contribute to this if you can help me with the testing, test cases are keep getting failed even on 

--- FAIL: TestScheduler_WithMonitor (0.59s)
    scheduler_test.go:39: found unexpected goroutines:

--- a/job.go
+++ b/job.go
@@ -115,8 +115,8 @@ type limitRunsTo struct {
 // implemented to provide a custom cron implementation for
 // the job. Pass in the implementation using the JobOption WithCronImplementation.
 type Cron interface {
-	IsValid(string) error
-	Next(string, time.Time) time.Time
+	IsValid(crontab string, location *time.Location, now time.Time) error
+	Next(lastRun time.Time) time.Time
 }
 
 // -----------------------------------------------

--- a/job.go
+++ b/job.go
@@ -107,15 +107,15 @@ type limitRunsTo struct {
 
 // -----------------------------------------------
 // -----------------------------------------------
-// --------------- Custom Cron ------------------
+// --------------- Custom Cron -------------------
 // -----------------------------------------------
 // -----------------------------------------------
 
 // Cron defines the interface that must be
-// implemented to create a cron.
-
+// implemented to provide a custom cron implementation for
+// the job. Pass in the implementation using the JobOption WithCronImplementation.
 type Cron interface {
-	IsValid(string) bool
+	IsValid(string) error
 	Next(string, time.Time) time.Time
 }
 
@@ -131,20 +131,20 @@ type JobDefinition interface {
 	setup(j *internalJob, l *time.Location, now time.Time) error
 }
 
-// Default cron implementation using robfig
+// Default cron implementation
 
 func newDefaultCronImplementation(withSeconds bool) Cron {
-	return &RobfigCron{
+	return &defaultCron{
 		withSeconds: withSeconds,
 	}
 }
 
-type RobfigCron struct {
+type defaultCron struct {
 	cronSchedule cron.Schedule
 	withSeconds  bool
 }
 
-func (r *RobfigCron) IsValid(crontab string) bool {
+func (r *defaultCron) IsValid(crontab string) error {
 	var withLocation string
 	if strings.HasPrefix(crontab, "TZ=") || strings.HasPrefix(crontab, "CRON_TZ=") {
 		withLocation = crontab
@@ -166,16 +166,16 @@ func (r *RobfigCron) IsValid(crontab string) bool {
 		cronSchedule, err = cron.ParseStandard(withLocation)
 	}
 	if err != nil {
-		return false
+		return errors.Join(ErrCronJobParse, err)
 	}
 	if cronSchedule.Next(time.Now()).IsZero() {
-		return false
+		return ErrCronJobInvalid
 	}
 	r.cronSchedule = cronSchedule
-	return true
+	return nil
 }
 
-func (r *RobfigCron) Next(crontab string, lastRun time.Time) time.Time {
+func (r *defaultCron) Next(crontab string, lastRun time.Time) time.Time {
 	return r.cronSchedule.Next(lastRun)
 }
 
@@ -192,7 +192,8 @@ func (c cronJobDefinition) setup(j *internalJob, location *time.Location, _ time
 		c.cron = j.cron
 	}
 
-	if !c.cron.IsValid(c.crontab) {
+	if err := c.cron.IsValid(c.crontab); err != nil {
+	    return err
 		return ErrCronJobInvalid
 	}
 
@@ -653,7 +654,8 @@ func WithName(name string) JobOption {
 	}
 }
 
-// JobOption to set custom Cron implementation
+// WithCronImplementation sets the custom Cron implementation for the job.
+// This is only utilized for the CronJob type.
 func WithCronImplementation(c Cron) JobOption {
 	return func(j *internalJob, _ time.Time) error {
 		j.cron = c

--- a/job.go
+++ b/job.go
@@ -106,6 +106,20 @@ type limitRunsTo struct {
 
 // -----------------------------------------------
 // -----------------------------------------------
+// --------------- Custom Cron ------------------
+// -----------------------------------------------
+// -----------------------------------------------
+
+// Cron defines the interface that must be
+// implemented to create a cron.
+
+type Cron interface {
+	IsValid() bool
+	Next(time.Time) time.Time
+}
+
+// -----------------------------------------------
+// -----------------------------------------------
 // --------------- Job Variants ------------------
 // -----------------------------------------------
 // -----------------------------------------------
@@ -116,21 +130,45 @@ type JobDefinition interface {
 	setup(j *internalJob, l *time.Location, now time.Time) error
 }
 
+var _ JobDefinition = (*customCron)(nil)
+
+type customCron struct {
+	crontab string
+	cron    Cron
+}
+
+func (c customCron) setup(j *internalJob, location *time.Location, now time.Time) error {
+	if !c.cron.IsValid() {
+		return ErrCronJobInvalid
+	}
+	j.jobSchedule = &cronJob{cronSchedule: c.cron}
+	return nil
+}
+
+func CustomCronJob(crontab string, cron Cron) JobDefinition {
+	return customCron{
+		crontab: crontab,
+		cron:    cron,
+	}
+}
+
+// default cron job implimentation
 var _ JobDefinition = (*cronJobDefinition)(nil)
 
 type cronJobDefinition struct {
-	crontab     string
-	withSeconds bool
+	crontab      string
+	withSeconds  bool
+	cronSchedule cron.Schedule
 }
 
-func (c cronJobDefinition) setup(j *internalJob, location *time.Location, now time.Time) error {
+func (c *cronJobDefinition) IsValid() bool {
 	var withLocation string
 	if strings.HasPrefix(c.crontab, "TZ=") || strings.HasPrefix(c.crontab, "CRON_TZ=") {
 		withLocation = c.crontab
 	} else {
 		// since the user didn't provide a timezone default to the location
 		// passed in by the scheduler. Default: time.Local
-		withLocation = fmt.Sprintf("CRON_TZ=%s %s", location.String(), c.crontab)
+		withLocation = fmt.Sprintf("CRON_TZ=%s %s", time.Local, c.crontab)
 	}
 
 	var (
@@ -145,13 +183,24 @@ func (c cronJobDefinition) setup(j *internalJob, location *time.Location, now ti
 		cronSchedule, err = cron.ParseStandard(withLocation)
 	}
 	if err != nil {
-		return errors.Join(ErrCronJobParse, err)
+		return false
 	}
-	if cronSchedule.Next(now).IsZero() {
+	if cronSchedule.Next(time.Now()).IsZero() {
+		return false
+	}
+	c.cronSchedule = cronSchedule
+	return true
+}
+
+func (c *cronJobDefinition) Next(lastRun time.Time) time.Time {
+	return c.cronSchedule.Next(lastRun)
+}
+
+func (c *cronJobDefinition) setup(j *internalJob, location *time.Location, _ time.Time) error {
+	if !c.IsValid() {
 		return ErrCronJobInvalid
 	}
-
-	j.jobSchedule = &cronJob{cronSchedule: cronSchedule}
+	j.jobSchedule = &cronJob{cronSchedule: c}
 	return nil
 }
 
@@ -162,7 +211,7 @@ func (c cronJobDefinition) setup(j *internalJob, location *time.Location, now ti
 // crontab in the form `TZ=America/Chicago * * * * *` or
 // `CRON_TZ=America/Chicago * * * * *`
 func CronJob(crontab string, withSeconds bool) JobDefinition {
-	return cronJobDefinition{
+	return &cronJobDefinition{
 		crontab:     crontab,
 		withSeconds: withSeconds,
 	}
@@ -818,7 +867,7 @@ type jobSchedule interface {
 var _ jobSchedule = (*cronJob)(nil)
 
 type cronJob struct {
-	cronSchedule cron.Schedule
+	cronSchedule Cron
 }
 
 func (j *cronJob) next(lastRun time.Time) time.Time {

--- a/job.go
+++ b/job.go
@@ -146,14 +146,14 @@ type defaultCron struct {
 	withSeconds  bool
 }
 
-func (r *defaultCron) IsValid(crontab string) error {
+func (r *defaultCron) IsValid(crontab string, location *time.Location, now time.Time) error {
 	var withLocation string
 	if strings.HasPrefix(crontab, "TZ=") || strings.HasPrefix(crontab, "CRON_TZ=") {
 		withLocation = crontab
 	} else {
 		// since the user didn't provide a timezone default to the location
 		// passed in by the scheduler. Default: time.Local
-		withLocation = fmt.Sprintf("CRON_TZ=%s %s", time.Local, crontab)
+		withLocation = fmt.Sprintf("CRON_TZ=%s %s", location.String(), crontab)
 	}
 
 	var (

--- a/job.go
+++ b/job.go
@@ -139,6 +139,8 @@ func newDefaultCronImplementation(withSeconds bool) Cron {
 	}
 }
 
+var _ Cron = (*defaultCron)(nil)
+
 type defaultCron struct {
 	cronSchedule cron.Schedule
 	withSeconds  bool
@@ -168,14 +170,14 @@ func (r *defaultCron) IsValid(crontab string) error {
 	if err != nil {
 		return errors.Join(ErrCronJobParse, err)
 	}
-	if cronSchedule.Next(time.Now()).IsZero() {
+	if cronSchedule.Next(now).IsZero() {
 		return ErrCronJobInvalid
 	}
 	r.cronSchedule = cronSchedule
 	return nil
 }
 
-func (r *defaultCron) Next(crontab string, lastRun time.Time) time.Time {
+func (r *defaultCron) Next(lastRun time.Time) time.Time {
 	return r.cronSchedule.Next(lastRun)
 }
 
@@ -187,14 +189,13 @@ type cronJobDefinition struct {
 	cron    Cron
 }
 
-func (c cronJobDefinition) setup(j *internalJob, location *time.Location, _ time.Time) error {
+func (c cronJobDefinition) setup(j *internalJob, location *time.Location, now time.Time) error {
 	if j.cron != nil {
 		c.cron = j.cron
 	}
 
-	if err := c.cron.IsValid(c.crontab); err != nil {
+	if err := c.cron.IsValid(c.crontab, location, now); err != nil {
 	    return err
-		return ErrCronJobInvalid
 	}
 
 	j.jobSchedule = &cronJob{crontab: c.crontab, cronSchedule: c.cron}

--- a/job.go
+++ b/job.go
@@ -879,7 +879,7 @@ type cronJob struct {
 }
 
 func (j *cronJob) next(lastRun time.Time) time.Time {
-	return j.cronSchedule.Next(j.crontab, lastRun)
+	return j.cronSchedule.Next(lastRun)
 }
 
 var _ jobSchedule = (*durationJob)(nil)

--- a/job.go
+++ b/job.go
@@ -24,6 +24,7 @@ type internalJob struct {
 	id        uuid.UUID
 	name      string
 	tags      []string
+	cron      Cron
 	jobSchedule
 
 	// as some jobs may queue up, it's possible to
@@ -114,8 +115,8 @@ type limitRunsTo struct {
 // implemented to create a cron.
 
 type Cron interface {
-	IsValid() bool
-	Next(time.Time) time.Time
+	IsValid(string) bool
+	Next(string, time.Time) time.Time
 }
 
 // -----------------------------------------------
@@ -130,45 +131,27 @@ type JobDefinition interface {
 	setup(j *internalJob, l *time.Location, now time.Time) error
 }
 
-var _ JobDefinition = (*customCron)(nil)
+// Default cron implementation using robfig
 
-type customCron struct {
-	crontab string
-	cron    Cron
-}
-
-func (c customCron) setup(j *internalJob, location *time.Location, now time.Time) error {
-	if !c.cron.IsValid() {
-		return ErrCronJobInvalid
-	}
-	j.jobSchedule = &cronJob{cronSchedule: c.cron}
-	return nil
-}
-
-func CustomCronJob(crontab string, cron Cron) JobDefinition {
-	return customCron{
-		crontab: crontab,
-		cron:    cron,
+func newDefaultCronImplementation(withSeconds bool) Cron {
+	return &RobfigCron{
+		withSeconds: withSeconds,
 	}
 }
 
-// default cron job implimentation
-var _ JobDefinition = (*cronJobDefinition)(nil)
-
-type cronJobDefinition struct {
-	crontab      string
-	withSeconds  bool
+type RobfigCron struct {
 	cronSchedule cron.Schedule
+	withSeconds  bool
 }
 
-func (c *cronJobDefinition) IsValid() bool {
+func (r *RobfigCron) IsValid(crontab string) bool {
 	var withLocation string
-	if strings.HasPrefix(c.crontab, "TZ=") || strings.HasPrefix(c.crontab, "CRON_TZ=") {
-		withLocation = c.crontab
+	if strings.HasPrefix(crontab, "TZ=") || strings.HasPrefix(crontab, "CRON_TZ=") {
+		withLocation = crontab
 	} else {
 		// since the user didn't provide a timezone default to the location
 		// passed in by the scheduler. Default: time.Local
-		withLocation = fmt.Sprintf("CRON_TZ=%s %s", time.Local, c.crontab)
+		withLocation = fmt.Sprintf("CRON_TZ=%s %s", time.Local, crontab)
 	}
 
 	var (
@@ -176,7 +159,7 @@ func (c *cronJobDefinition) IsValid() bool {
 		err          error
 	)
 
-	if c.withSeconds {
+	if r.withSeconds {
 		p := cron.NewParser(cron.SecondOptional | cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow | cron.Descriptor)
 		cronSchedule, err = p.Parse(withLocation)
 	} else {
@@ -188,19 +171,32 @@ func (c *cronJobDefinition) IsValid() bool {
 	if cronSchedule.Next(time.Now()).IsZero() {
 		return false
 	}
-	c.cronSchedule = cronSchedule
+	r.cronSchedule = cronSchedule
 	return true
 }
 
-func (c *cronJobDefinition) Next(lastRun time.Time) time.Time {
-	return c.cronSchedule.Next(lastRun)
+func (r *RobfigCron) Next(crontab string, lastRun time.Time) time.Time {
+	return r.cronSchedule.Next(lastRun)
 }
 
-func (c *cronJobDefinition) setup(j *internalJob, location *time.Location, _ time.Time) error {
-	if !c.IsValid() {
+// default cron job implimentation
+var _ JobDefinition = (*cronJobDefinition)(nil)
+
+type cronJobDefinition struct {
+	crontab string
+	cron    Cron
+}
+
+func (c cronJobDefinition) setup(j *internalJob, location *time.Location, _ time.Time) error {
+	if j.cron != nil {
+		c.cron = j.cron
+	}
+
+	if !c.cron.IsValid(c.crontab) {
 		return ErrCronJobInvalid
 	}
-	j.jobSchedule = &cronJob{cronSchedule: c}
+
+	j.jobSchedule = &cronJob{crontab: c.crontab, cronSchedule: c.cron}
 	return nil
 }
 
@@ -211,9 +207,9 @@ func (c *cronJobDefinition) setup(j *internalJob, location *time.Location, _ tim
 // crontab in the form `TZ=America/Chicago * * * * *` or
 // `CRON_TZ=America/Chicago * * * * *`
 func CronJob(crontab string, withSeconds bool) JobDefinition {
-	return &cronJobDefinition{
-		crontab:     crontab,
-		withSeconds: withSeconds,
+	return cronJobDefinition{
+		crontab: crontab,
+		cron:    newDefaultCronImplementation(withSeconds),
 	}
 }
 
@@ -657,6 +653,14 @@ func WithName(name string) JobOption {
 	}
 }
 
+// JobOption to set custom Cron implementation
+func WithCronImplementation(c Cron) JobOption {
+	return func(j *internalJob, _ time.Time) error {
+		j.cron = c
+		return nil
+	}
+}
+
 // WithSingletonMode keeps the job from running again if it is already running.
 // This is useful for jobs that should not overlap, and that occasionally
 // (but not consistently) run longer than the interval between job runs.
@@ -867,11 +871,12 @@ type jobSchedule interface {
 var _ jobSchedule = (*cronJob)(nil)
 
 type cronJob struct {
+	crontab      string
 	cronSchedule Cron
 }
 
 func (j *cronJob) next(lastRun time.Time) time.Time {
-	return j.cronSchedule.Next(lastRun)
+	return j.cronSchedule.Next(j.crontab, lastRun)
 }
 
 var _ jobSchedule = (*durationJob)(nil)

